.github/actions/slack-message/action.yml

```diff
@@ -0,0 +1,38 @@
+name: Send a message to slack
+
+inputs:
+  commit_url:
+    description: normally "${{ github.server_url }}/${{ github.repository }}/commit/${{ GITHUB_SHA}}"
+  job_status:
+    description: passthrough for job.status
+  msg:
+    description: The slack message text
+  run_url:
+    description: normally "${{ GITHUB_REF#refs/heads/ }}-${{ GITHUB_SHA }}"
+  sha:
+    description: normally "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+  slack_url:
+    description: passthrough for secrets.SLACK_WEBHOOK_URL
+
+runs:
+  using: composite
```

Comment:
is this not able to be a reusable workflow? According to some [random blog](https://dev.to/n3wt0n/composite-actions-vs-reusable-workflows-what-is-the-difference-github-actions-11kd) reusable workflows can consume secrets

Reply:
It could be.  Personally, I like the separation produced by making smaller, one-off actions composites, and leaving workflows for larger, more involved compositions.

Reply:
honestly it seems simpler to have all of one type if the differences between composite and reusable workflows are small. I spent a good 5-10min digging around for decent info on what a composite workflow is just to review this PR. There's no need to cause further confusion especially with there being little documentation on some stuff

Reply:
With such a low rate of change for CI tooling stuffs, I don't think the difference adds real burden for development.  For the most part these things are highly stable, with the odd addition of a composite action.  And the composites now have good boilerplate to work off of, both for creation and usage.